apps/core/client/mutations/submit-change-password.ts

```diff
@@ -3,17 +3,25 @@ import { z } from 'zod';
 import { client } from '..';
 import { graphql } from '../graphql';
 
-export const ChangePasswordSchema = z
-  .object({
-    customerId: z.string(),
-    customerToken: z.string(),
-    newPassword: z.string(),
-    confirmPassword: z.string(),
-  })
-  .required();
+const ChangePasswordFieldsSchema = z.object({
+  customerId: z.string(),
+  customerToken: z.string(),
+  currentPassword: z.string().min(1),
+  newPassword: z.string().min(1),
+  confirmPassword: z.string().min(1),
+});
+
+export const CustomerChangePasswordSchema = ChangePasswordFieldsSchema.omit({
+  customerId: true,
+  customerToken: true,
+});
+
+export const ChangePasswordSchema = ChangePasswordFieldsSchema.omit({
+  currentPassword: true,
+}).required();
```

Comment:
Yeah, I see your point. But I guess it can be on tricky side - i use `Schema` in `<Form/>` and in server `action`. If I define `Schema` into `<Form/>`, then I got error in `submit-customer-change-password-form` action on submitting:
<img width="1109" alt="Screenshot 2024-04-16 at 14 48 19" src="https://github.com/bigcommerce/catalyst/assets/67792608/804a1789-3563-4ba8-a019-4b34233ca92b">

If I move `Schema` to server action then I get error in `<Form/>` on handling User input like this:
 
<img width="963" alt="Screenshot 2024-04-16 at 14 53 00" src="https://github.com/bigcommerce/catalyst/assets/67792608/1c652d52-fdc8-44d0-8c1d-eb844e5de1e3">

What do you say on this? maybe I missed smth...

Reply:
This is fine then!